backend/typescript/rest/behaviourRoutes.ts

```diff
@@ -52,7 +52,11 @@ behaviourRouter.get("/:id", async (req, res) => {
     const behaviour = await behaviourService.getBehaviour(id);
     res.status(200).json(behaviour);
   } catch (e: unknown) {
-    res.status(500).send(getErrorMessage(e));
+    if (e instanceof NotFoundError) {
+      res.status(404).send(getErrorMessage(e));
+    } else {
+      res.status(500).send(getErrorMessage(e));
```

Comment:
for 500 i dont think we want to send the error message back. 500 is internal error so ideally we want to log the error internally, confirm @jerry-cheng5

Reply:
Yea I agree, I think that's the proper way to handle 500 errors. 

(Normally its for security and shi and we don't reaaally have to worry about that, but its always good to practice the proper response conventions)

Reply:
So maybe just send some generic message back like:

An unexpected internal server error occurred. Please try again later.

or

An unexpected error occurred on our end. Please try again later.